tools/setup_generation.py

```diff
@@ -513,8 +514,22 @@ def format_getting_started_navigation(filepath: str) -> str:
 getting_started_navigation = "\n".join(step_folders) + '\n'
 
 
+
+
+# ------------------------------------------------------------------------
+# Step 4
+#   Generating site url
+# ------------------------------------------------------------------------
+print("Step 4/4: Generating the Site URL")
+current_branch_name = subprocess.run(['git', 'branch', '--show-current'], stdout=subprocess.PIPE).stdout.decode()[:-1]
```

Comment:
That is slightly dangerous.
The current branch (from taipy-doc) may be different than a real version number, like in a situation of a fix.
This could be better handled with a program argument. May be both.

Reply:
Putting the version in the mkdocs file is a good way to do it?
```
extra:
  version: 1.0
```